pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/FunctionsBase.java

```diff
@@ -94,14 +183,106 @@ public void registerFunction(final @PathParam("tenant") String tenant,
     })
     @Path("/{tenant}/{namespace}/{functionName}")
     @Consumes(MediaType.MULTIPART_FORM_DATA)
-    public void updateFunction(final @PathParam("tenant") String tenant,
-                               final @PathParam("namespace") String namespace,
-                               final @PathParam("functionName") String functionName,
-                               final @FormDataParam("data") InputStream uploadedInputStream,
-                               final @FormDataParam("data") FormDataContentDisposition fileDetail,
-                               final @FormDataParam("url") String functionPkgUrl,
-                               final @FormDataParam("functionConfig") String functionConfigJson,
-                               final @FormDataParam("updateOptions") UpdateOptions updateOptions) throws IOException {
+    public void updateFunction(
+            @ApiParam(value = "The function's tenant")
+            final @PathParam("tenant") String tenant,
+            @ApiParam(value = "The function's namespace")
+            final @PathParam("namespace") String namespace,
+            @ApiParam(value = "The function's name")
+            final @PathParam("functionName") String functionName,
+            final @FormDataParam("data") InputStream uploadedInputStream,
+            final @FormDataParam("data") FormDataContentDisposition fileDetail,
+            final @FormDataParam("url") String functionPkgUrl,
+            @ApiParam(
+                    value = "A JSON value presenting a functions config playload. An example of the expected functions can be found down here.  \n" +
+                            "autoAck  \n" +
+                            "  Whether or not the framework will automatically acknowledge messages  \n" +
+                            "runtime  \n" +
+                            "  What is the runtime of the function. Possible Values: [JAVA, PYTHON, GO]  \n" +
+                            "resources  \n" +
+                            "  The size of the system resources allowed by the function runtime, include: cpu, ram, disk.  \n" +
```

Comment:
The size of the system resources allowed by the function runtime. System resources include CPU, ram, and disk.